src/targets/gpu/hip.cpp

```diff
@@ -47,7 +47,14 @@ MIGRAPHX_REGISTER_OP(hip_copy_literal)
 using hip_ptr      = MIGRAPHX_MANAGE_PTR(void, hipFree);
 using hip_host_ptr = MIGRAPHX_MANAGE_PTR(void, hipHostUnregister);
 
-std::string hip_error(int error) { return hipGetErrorString(static_cast<hipError_t>(error)); }
+std::string hip_error(int error)
+{
+    // reset HIP's internal status to hipSuccess
+    auto last_error = hipGetLastError();
+    assert(last_error == error);
```

Comment:
I dont think we should have side effects in the function. It should only convert an error id to a string regardless of where it came from.

Reply:
I don't see it as a side-effect. 

If `hip_error()` is called two times back to back on two unsuccessful calls, it would generate correct error message. 

Without side effect, it can't.

Reply:
should the side effect be made explicit  then?

Reply:
By explicit, it should be a call made to `hipGetLastError` outside of the `hip_error` function.